More escape characters added

RCEMaddox opened the pull request:
Added some additional special characters to the backend that will now be properly escaped.

7RedViolin commented:
Hey @RCEMaddox - thanks for contributing to the translator! Before this can be merged, can you update the failing tests to account for the changes in escape characters?

Thanks!

RCEMaddox commented:
Reviewing the tests that were failing, I found that CbR and CbC have different sets of characters that are escaped. The following changes have been made in this commit:

* Created an `apply_cbr_pipeline_attrs` method, permitting different character sets to be escaped dependent on the pipeline  leveraged and field being transformed.
*  Modified some code in `convert_value_str` to prevent hyphens (-) to be erroneously escaped in CbR
* Overrode method `convert_condition_field_eq_val_str` in order to pass `field` value, a necessary argument in the new `apply_cbr_pipeline_attrs` method.
* Modified tests to account for new escape logic.

Thank you for looking this over, and if there's anything else that's needed, please let me know.

RCEMaddox commented:
Hey @7RedViolin, bumping this PR in case it slipped through. Thanks again!
